ethapi/api.go

```diff
@@ -799,7 +799,7 @@ func (s *PublicBlockChainAPI) GetHeaderByHash(ctx context.Context, hash common.H
 
 func (s *PublicBlockChainAPI) getBlockReceipts(ctx context.Context, blkNumber rpc.BlockNumber) (types.Receipts, error) {
 	if blkNumber == rpc.EarliestBlockNumber {
-		return nil, nil
+		return types.Receipts{}, nil
```

Comment:
is this because `nil` would be interpreted as `skipped`?

Reply:
The RPC handler consuming this make a difference between an empty list of receipts and the `nil` list when encoding the receipts-root-hash into a header. A `nil` list results in a 0-root hash, while a empty list results in the hash of an empty list.

This is not related to skipped transactions.